debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorIT.java

```diff
@@ -2767,6 +2769,211 @@ public void shouldNotAckLsnOnSource() throws Exception {
+                slotAfter.slotLastFlushedLsn(), postActivityServerLsn);
+
+        assertThat(slotAfter.slotLastFlushedLsn())
+                .describedAs("LSN should not advanced due to unmonitored activity in connector mode")
+                .isEqualTo(slotBefore.slotLastFlushedLsn());
+        assertThat(slotAfter.slotLastFlushedLsn())
+                .describedAs("Slot LSN should be behind server LSN as pgjdbc keep alive flushing is disabled")
+                .isLessThan(postActivityServerLsn);
+
+        stopConnector();
+    }
+
+    @Test
+    @FixFor("DBZ-9641")
+    public void shouldFlushLsnOfUnmonitoredActivityInConnectorAndDriverMode() throws Exception {
+        String slotName = "pgjdbc_keepalive_slot";
+        String pluginName = "pgoutput";
+        TestHelper.dropReplicationSlot(slotName);
+        TestHelper.execute(SETUP_TABLES_STMT);
+
+        final Configuration.Builder configBuilder = TestHelper.defaultConfig()
+                .with(PostgresConnectorConfig.SLOT_NAME, slotName)
+                .with(PostgresConnectorConfig.PLUGIN_NAME, pluginName)
+                .with(PostgresConnectorConfig.LSN_FLUSH_MODE, PostgresConnectorConfig.LsnFlushMode.CONNECTOR_AND_DRIVER.getValue())
+                .with(PostgresConnectorConfig.SCHEMA_INCLUDE_LIST, "s1")
+                .with(PostgresConnectorConfig.STATUS_UPDATE_INTERVAL_MS, 100)
+                .with(PostgresConnectorConfig.SNAPSHOT_MODE, SnapshotMode.NO_DATA);
+
+        start(PostgresConnector.class, configBuilder.build());
+        assertConnectorIsRunning();
+        waitForStreamingRunning();
+
+        final SlotState slotBefore = getReplicationSlot(slotName, pluginName);
+
+        Lsn preActivityServerLsn;
+        try (PostgresConnection conn = TestHelper.create()) {
+            long walLocation = conn.currentXLogLocation();
+            preActivityServerLsn = Lsn.valueOf(walLocation);
+        }
+        // Generate unmonitored WAL activity
+        TestHelper.execute("CHECKPOINT;");
+        TestHelper.execute("SELECT pg_switch_wal();");
+        TestHelper.execute("CHECKPOINT;");
+        TestHelper.execute("SELECT pg_switch_wal();");
+
+        Lsn postActivityServerLsn;
+        try (PostgresConnection conn = TestHelper.create()) {
+            long walLocation = conn.currentXLogLocation();
+            postActivityServerLsn = Lsn.valueOf(walLocation);
+        }
+
+        Awaitility.await().atMost(5, TimeUnit.SECONDS).pollInterval(100, TimeUnit.MILLISECONDS).untilAsserted(() -> {
+            assertThat(postActivityServerLsn)
+                    .describedAs("Physical WAL activity should have advanced the server LSN")
+                    .isGreaterThan(preActivityServerLsn);
+        });
+
+        Awaitility.await().atMost(Duration.ofSeconds(65)) // Completes in low seconds, but we should wait > 60 seconds for wal_sender_timeout
```

Comment:
Can we get this value dynamically from the server? Also does it make sense to change the current configuration to a smaller timeout?